crates/paimon/src/fs/seekable_input.rs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use async_trait::async_trait;
+use std::io::{self, SeekFrom};
+use tokio::io::{AsyncRead, AsyncSeek};
+
+#[async_trait]
+pub trait SeekableInputStream: AsyncRead + AsyncSeek + Unpin {
```

Comment:
I suggest not introducing traits like this. We should avoid using anything that accepts or consumes `AsyncSeek`.

We can start a seperate PR for this to add IO trait in paimon.

Reply:
Of course, I'll start with the changes to the IO module. I really appreciate your helpful suggestions.